### PR TITLE
load user list by orga admin role

### PR DESCRIPTION
**Pull request by jalalifardemos**
### Ticket: https://demoseurope.youtrack.cloud/issue/DPLAN-12132/Nutzende-konnen-auf-der-Seite-Nutzende-verwalten-nicht-geladen-werden

Description: 
- set new Permission ('feature_organisation_user_list') to access to the user list and its fields
- replace this Permission with ('area_organisation') in vue
- use Permission in js
- use Permission in different resourceTypes
- set Permission in Core for three Roles (support, orga, customer_master)
- use Permission in api 1 controller to give back list of Orgas

Delete the checkbox if it doesn't apply/isn't necessary.

- [x] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php [hunk 1]
```diff
@@ -54,7 +54,7 @@ public function isGetAllowed(): bool
 
     public function isListAllowed(): bool
     {
-        return false;
+        return $this->isAvailable();
```
Comment (sierrazamudiodemos) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php:57]: I would write true directly

Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php:57]: Why? In every resourcetype we use this method for true

Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php:57]: i think this should be something like return $this->currentUser->hasPermission('feature_user_list');

**Review by bakridemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php (hunk 1 above)
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php:57]: this permission allows the current user to fetch all users the current user is allowed to see from the JSON:API route, it's seems to be enabled everywhere (most of time for the support Role), but i'm honestly nt 100% sure, so it make sense for me to use here to check when the list should be allowed or not, may be u can check it too just to be sure

Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php]: done in [8b33081](https://github.com/demos-europe/demosplan-core/pull/3475/commits/8b33081bee61d3594962747f4e2bacfbfa411568)

Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php]: sorry my bad i thought this is the user resource type @sierrazamudiodemos u re right it works for the support role so it has to be a permissions problem

File: config/permissions.yml [hunk 2]
```diff
@@ -1340,6 +1340,11 @@ feature_organisation_set_showlist:
     label: 'Anzeigen der Organisation in der Institutionen-Liste'
     loginRequired: true
     parent: [area_manage_orgas, area_manage_orgas_all]
+feature_organisation_user_list:
+    description: 'Show Organisations fields in use list'
```
Comment (bakridemos) [inline config/permissions.yml:1344]: may be we have to explain why we need this new permission, something like 'unlike the 'area_organisations' permission, the 'feature_organisation_user_list' permission allows to show Organisations list but does not allow access to the 'Organisation' item in the menu.

File: demosplan/DemosPlanCoreBundle/Permissions/Permissions.php [hunk 3]
```diff
@@ -279,6 +279,7 @@ protected function setGlobalPermissions(): void
                 'feature_json_api_update',
                 'feature_orga_edit_all_fields',
                 'feature_orga_get',
+                'feature_organisation_user_list',
```
Comment (bakridemos) [inline demosplan/DemosPlanCoreBundle/Permissions/Permissions.php:282]: may be we can comment this too something like 'im Gegensatz zur Berechtigung „area_organisations“ ermöglicht die Berechtigung „feature_organisation_user_list“ die Anzeige der Organisationsliste, gewährt aber keinen Zugriff auf das Element „Organisation“ im Menü'

File: config/permissions.yml (hunk 2 above)
Comment (jalalifardemos) [reply config/permissions.yml]: done in [175abdd](https://github.com/demos-europe/demosplan-core/pull/3475/commits/175abddb14582d545653d01d87448040e005a1f8)

File: demosplan/DemosPlanCoreBundle/Permissions/Permissions.php (hunk 3 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Permissions/Permissions.php:284]: done in [1abaffb](https://github.com/demos-europe/demosplan-core/pull/3475/commits/1abaffb4f36a33fd142e6a0a6e1acaeb8395060f)

**Review by bakridemos — APPROVED** [review]
some comments but looks good to me

File: client/js/components/user/DpUserList/DpUserFormFields.vue [hunk 4]
```diff
@@ -271,7 +271,7 @@ export default {
     },
 
     isManagingSingleOrganisation () {
-      return hasPermission('area_organisations') === false && this.presetUserOrgaId
+      return hasPermission('feature_organisation_user_list') === false && this.presetUserOrgaId
```
Comment (salisdemos) [inline client/js/components/user/DpUserList/DpUserFormFields.vue:274]:
```suggestion
      return !hasPermission('feature_organisation_user_list') && this.presetUserOrgaId
```

File: client/js/components/user/DpUserList/DpUserFormFields.vue [hunk 5]
```diff
@@ -379,7 +379,7 @@ export default {
     },
 
     resetData () {
-      if (hasPermission('area_organisations') === false) {
+      if (hasPermission('feature_organisation_user_list') === false) {
```
Comment (salisdemos) [inline client/js/components/user/DpUserList/DpUserFormFields.vue:382]:
```suggestion
      if (!hasPermission('feature_organisation_user_list')) {
```

**Review by salisdemos — CHANGES_REQUESTED** [review]
just nitpicks

File: client/js/components/user/DpUserList/DpUserFormFields.vue (hunk 5 above)
Comment (jalalifardemos) [reply client/js/components/user/DpUserList/DpUserFormFields.vue]: done in [41b9b0c](https://github.com/demos-europe/demosplan-core/pull/3475/commits/41b9b0cc616c8556a288e1c5d05e8b4613494e1b)

File: client/js/components/user/DpUserList/DpUserFormFields.vue (hunk 4 above)
Comment (jalalifardemos) [reply client/js/components/user/DpUserList/DpUserFormFields.vue]: done in [41b9b0c](https://github.com/demos-europe/demosplan-core/pull/3475/commits/41b9b0cc616c8556a288e1c5d05e8b4613494e1b)

**Review by salisdemos — APPROVED** [review]

**Review by sierrazamudiodemos — APPROVED** [review]

**Pull request merged**